packages/react/src/components/Storage/FileUploader/FileUploader.tsx

```diff
@@ -61,33 +41,55 @@ export function FileUploader({
     if (!event.target.files || event.target.files.length === 0) return;
 
     const { files } = event.target;
-    setFiles([...files]);
+    setTargetFiles(files);
+
     setShowPreviewer(true);
   };
+
+  const onClose = () => {
+    setShowPreviewer(false);
+  };
+
   const CommonProps = {
     acceptedFileTypes,
     multiple,
     onFileChange,
   };
 
   if (showPreviewer) {
-    return <Previewer />;
+    return (
+      <Previewer
+        acceptedFileTypes={acceptedFileTypes}
+        fileNames={fileNames}
+        files={files}
+        inDropZone={inDropZone}
+        level={level}
+        multiple={multiple}
+        onClose={onClose}
+        onDragEnter={onDragEnter}
+        onDragLeave={onDragLeave}
+        onDragOver={onDragOver}
+        onDragStart={onDragStart}
+        onDrop={onDrop}
+        onFileChange={onFileChange}
+      />
+    );
   } else if (variation === 'button') {
```

Comment:
nit: you don't really need the else statements since each if returns. It actually reads quite odd, since `showPreviewer` and  `variation === 'button'` don't seem to have much in common with each other.

Reply:
I can split it out into 2 different `if` statements, instead of using the `else if` pattern.